### PR TITLE
feat(api-v2): Add metadata routes (DSP-662)

### PR DESCRIPTION
**Pull request by benjamingeer**
Resolves DSP-662.
- [x] Refactor the plumbing in HttpTriplestoreConnector and TriplestoreMessages.scala
- [x]  Refactor RouteUtilV2 and KnoraResponseV2 to handle this new kind of response message containing a Turtle string
- [x] Implement two-way conversion between JSON-LD and Turtle
- [x]  Implement the metadata routes and messages
- [x] Implement `metadataResponderV2` 
- [x] Add documentation

SHACL validation will be added in a later pull request.

The processing of Turtle requests and responses is a first draft, and will be refactored later for DSP-902.

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/responders/ResponderManager.scala [hunk 1]
```diff
@@ -185,60 +186,70 @@ class ResponderManager(appActor: ActorRef) extends Actor with ActorLogging {
...
+    protected val searchResponderV2: SearchResponderV2 = makeDefaultSearchResponderV2
+
     /**
      * Constructs the default [[ResourcesResponderV2]].
      */
     protected final def makeDefaultResourcesResponderV2: ResourcesResponderV2 = new ResourcesResponderV2(responderData)
 
     /**
-     * Constructs the default [[ValuesResponderV2]].
+     * Subclasses of the can override this member to substitute a with a custom implementation instead of the default resources responder.
```
Comment (subotic) [inline webapi/src/main/scala/org/knora/webapi/responders/ResponderManager.scala:210]: What does that mean?

File: webapi/src/test/scala/org/knora/webapi/e2e/v2/BUILD.bazel [hunk 2]
```diff
@@ -208,3 +208,22 @@ scala_test(
...
+        "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = ALL_WEBAPI_MAIN_DEPENDENCIES + [
+        "//webapi:main_library",
+        "//webapi:test_library",
+        "@maven//:org_apache_jena_apache_jena_libs",
+    ] + BASE_TEST_DEPENDENCIES_WITH_JSON,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/e2e/v2/BUILD.bazel:229]: Missing newline

File: webapi/src/test/scala/org/knora/webapi/messages/v2/responder/metadatamessages/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = ALL_WEBAPI_MAIN_DEPENDENCIES + [
+        "//webapi:main_library",
+        "//webapi:test_library",
+         "@maven//:org_apache_jena_apache_jena_libs",
+    ] + BASE_TEST_DEPENDENCIES,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/v2/responder/metadatamessages/BUILD.bazel:23]: Missing newline

File: webapi/src/test/scala/org/knora/webapi/responders/v2/BUILD.bazel [hunk 4]
```diff
@@ -126,3 +126,22 @@ scala_test(
...
+            "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = ALL_WEBAPI_MAIN_DEPENDENCIES + [
+        "//webapi:main_library",
+        "//webapi:test_library",
+        "@maven//:org_apache_jena_apache_jena_libs",
+    ] + BASE_TEST_DEPENDENCIES,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/responders/v2/BUILD.bazel:147]: Missing newline

File: webapi/src/test/scala/org/knora/webapi/store/triplestore/AllTriplestoreSpec.scala [hunk 5]
```diff
@@ -109,6 +110,16 @@ class AllTriplestoreSpec extends CoreSpec(AllTriplestoreSpec.config) with Implic
         }
         """
 
+    val graphDataContent: String =
+        """
+        prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+        prefix jedi: <http://jedi.org/#>
+
+        <http://luke> jedi:tries "force for the first time" ;
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/store/triplestore/AllTriplestoreSpec.scala:118]: :-)

**Review by subotic — APPROVED** [review]
LGTM, Thanks!

The only thing that is missing is the route for getting all metadata of all projects. This can be done in another PR. Please open a YouTrack issue. Thanks!

**Comment by benjamingeer** [comment]
@subotic Thanks for the review!

> The only thing that is missing is the route for getting all metadata of all projects.

I'm concerned that when we have hundreds of projects, returning all metadata for all projects in one API response could mean returning a huge API response that could put too much load on the triplestore. Let's talk about it first.

File: webapi/src/main/scala/org/knora/webapi/responders/ResponderManager.scala (hunk 1 above)
Comment (benjamingeer) [reply webapi/src/main/scala/org/knora/webapi/responders/ResponderManager.scala]: The same typo was in all the existing comments in `ResponderManager`, we just copied and pasted it. :) Fixed now.

**Pull request merged**